lib/tests/streamlit/components_test.py

```diff
@@ -250,17 +253,79 @@ def test_json_and_df_args(self):
         self.assertIsNotNone(proto.args_dataframe)
 
     def test_key(self):
-        """Test the 'key' param"""
-        # TODO!
-        pass
+        """Two components with different arguments but the same `key` should have the same ID"""
+        self.test_component(foo="bar", key="baz")
+        proto1 = self.get_delta_from_queue().new_element.component_instance
+
+        # Ignore the raised exception to see if the IDs are the same.
+        try:
+            self.test_component(key="baz")
+        except DuplicateWidgetID:
+            pass
+
+        proto2 = self.get_delta_from_queue().new_element.component_instance
+        self.assertEqual(proto1.id, proto2.id)
```

Comment:
I might be misunderstanding, but I think this actually doesn't test what it claims to test.

The second call to `test_component` with the same key won't result in a new delta being enqueued, because an exception gets thrown. So `proto1` and `proto2` are actually the same object.

Instead, you just want to assert that the exception is raised for the second call to `test_component`:

```python
with self.assertRaises(DuplicateWidgetID):
    self.test_component(key="baz")
```

And no need to check that the IDs are the same - the exception being thrown is proof of that.

Reply:
I see. Thanks for the insight.

Reply:
Done!